lib/active_shipping/shipping/carriers/usps.rb

```diff
@@ -578,22 +582,25 @@ def parse_tracking_with_fields_response(response)
           success = !(tracking_info.elements['Error'] || summary_no_record?(tracking_summary))
           message = response_message(tracking_info)
           shipment_events = []
-          status, expected_delivery_date, actual_delivery_date = nil
+          status, status_category, expected_delivery_date, actual_delivery_date = nil, nil, nil, nil
 
           if success 
+            previous_event_time = Time.now
             tracking_details.each do |event|
-              ship_event = extract_track_with_fields_event(event)
-              shipment_events << ship_event if !ship_event.nil?
+              ship_event = extract_track_with_fields_event(event, previous_event_time)
+              previous_event_time = ship_event.time
+              shipment_events << ship_event
```

Comment:
Looks like you dropped the if nil check, is it now impossible for ship_event to be nil?